In [1]:
import pandas as pd


In [2]:
legodata=pd.read_csv('lego_sets.csv')

In [3]:
legodata.head(5)



ages  list_price  num_reviews  piece_count  play_star_rating  \
0  6-12       29.99          2.0        277.0               4.0   
1  6-12       19.99          2.0        168.0               4.0   
2  6-12       12.99         11.0         74.0               4.3   
3   12+       99.99         23.0       1032.0               3.6   
4   12+       79.99         14.0        744.0               3.2   

                                           prod_desc  prod_id  \
0  Catapult into action and take back the eggs fr...  75823.0   
1  Launch a flying attack and rescue the eggs fro...  75822.0   
2  Chase the piggy with lightning-fast Chuck and ...  75821.0   
3  Explore the architecture of the United States ...  21030.0   
4  Recreate the Solomon R. Guggenheim Museum® wit...  21035.0   

                                      prod_long_desc review_difficulty  \
0  Use the staircase catapult to launch Red into ...           Average   
1  Pilot Pig has taken off from Bird Island with ...              Easy   
2  Pitch speedy bird Chuck against the Piggy Car....              Easy   
3  Discover the architectural secrets of the icon...           Average   
4  Discover the architectural secrets of Frank Ll...       Challenging   

                         set_name  star_rating    theme_name  val_star_rating  \
0           Bird Island Egg Heist          4.5  Angry Birds™              4.0   
1              Piggy Plane Attack          5.0  Angry Birds™              4.0   
2                Piggy Car Escape          4.3  Angry Birds™              4.1   
3  United States Capitol Building          4.6  Architecture              4.3   
4   Solomon R. Guggenheim Museum®          4.6  Architecture              4.1   

  country  
0      US  
1      US  
2      US  
3      US  
4      US

In [4]:
# checking what type of values there is in the data set

legodata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12261 entries, 0 to 12260
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   ages               12261 non-null  object 
 1   list_price         12261 non-null  float64
 2   num_reviews        10641 non-null  float64
 3   piece_count        12261 non-null  float64
 4   play_star_rating   10486 non-null  float64
 5   prod_desc          11884 non-null  object 
 6   prod_id            12261 non-null  float64
 7   prod_long_desc     12261 non-null  object 
 8   review_difficulty  10206 non-null  object 
 9   set_name           12261 non-null  object 
 10  star_rating        10641 non-null  float64
 11  theme_name         12258 non-null  object 
 12  val_star_rating    10466 non-null  float64
 13  country            12261 non-null  object 
dtypes: float64(7), object(7)
memory usage: 1.3+ MB


In [5]:
#checking for missing data
legodata.isna().sum()

ages                    0
list_price              0
num_reviews          1620
piece_count             0
play_star_rating     1775
prod_desc             377
prod_id                 0
prod_long_desc          0
review_difficulty    2055
set_name                0
star_rating          1620
theme_name              3
val_star_rating      1795
country                 0
dtype: int64

In [6]:
#dropping num_reviews missing values throughout all data

legodata = legodata.loc[legodata['num_reviews'].notna()]

In [7]:
#checking missing values again
legodata.isna().sum()

ages                   0
list_price             0
num_reviews            0
piece_count            0
play_star_rating     155
prod_desc            293
prod_id                0
prod_long_desc         0
review_difficulty    435
set_name               0
star_rating            0
theme_name             3
val_star_rating      175
country                0
dtype: int64

In [8]:
#removing the rest of missing values
legodata = legodata.dropna()

In [9]:
legodata.isna().sum()

ages                 0
list_price           0
num_reviews          0
piece_count          0
play_star_rating     0
prod_desc            0
prod_id              0
prod_long_desc       0
review_difficulty    0
set_name             0
star_rating          0
theme_name           0
val_star_rating      0
country              0
dtype: int64

In [10]:
#checking how many different values data frame columns have:

for col in legodata.columns:
    print(col, '-', len(pd.unique(legodata[col])))


ages - 30
list_price - 675
num_reviews - 86
piece_count - 412
play_star_rating - 30
prod_desc - 626
prod_id - 620
prod_long_desc - 848
review_difficulty - 5
set_name - 619
star_rating - 27
theme_name - 37
val_star_rating - 33
country - 21


In [11]:
legodata.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9910 entries, 0 to 12260
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   ages               9910 non-null   object 
 1   list_price         9910 non-null   float64
 2   num_reviews        9910 non-null   float64
 3   piece_count        9910 non-null   float64
 4   play_star_rating   9910 non-null   float64
 5   prod_desc          9910 non-null   object 
 6   prod_id            9910 non-null   float64
 7   prod_long_desc     9910 non-null   object 
 8   review_difficulty  9910 non-null   object 
 9   set_name           9910 non-null   object 
 10  star_rating        9910 non-null   float64
 11  theme_name         9910 non-null   object 
 12  val_star_rating    9910 non-null   float64
 13  country            9910 non-null   object 
dtypes: float64(7), object(7)
memory usage: 1.1+ MB


In [12]:
#checking types of unique values in 'review_difficulty' column

legodata['review_difficulty'].unique()

array(['Average', 'Easy', 'Challenging', 'Very Easy', 'Very Challenging'],
      dtype=object)

In [13]:
#checking types of unique values in 'ages' column

legodata["ages"].unique()

array(['6-12', '12+', '7-12', '10+', '8-12', '5-12', '4-99', '4+', '9-12',
       '16+', '14+', '9-14', '7-14', '8-14', '4-7', '6+', '2-5', '1½-5',
       '1½-3', '9+', '5-8', '8+', '6-14', '5+', '10-16', '10-14', '11-16',
       '12-16', '9-16', '7+'], dtype=object)

Ages column has two tyoes of ages calculation: <br> age between two numbers (ex. 6-12), <br> age from certain number and up (ex. 7+)

After data wrangling process legodata has 9910 entries. <br> Total 14 columns from which seven columns are numeric values, and seven  - categorical.